apps/studio/src/hooks/useSearchQuery.ts

```diff
@@ -0,0 +1,62 @@
+import { useCallback, useState } from "react"
+import { useDebounce } from "@uidotdev/usehooks"
+
+import type { SearchResultResource } from "~/server/modules/resource/resource.types"
+import type { ResourceType } from "~prisma/generated/generatedEnums"
+import { trpc } from "~/utils/trpc"
+
+interface UseSearchQueryProps {
+  siteId: string
+  resourceTypes: ResourceType[]
+  onSearchSuccess?: () => void
+}
+export const useSearchQuery = ({
+  siteId,
+  resourceTypes,
+  onSearchSuccess,
+}: UseSearchQueryProps) => {
+  const [searchValue, setSearchValue] = useState("")
+  const debouncedSearchTerm = useDebounce(searchValue, 300)
+
+  const { data, isLoading } = useCallback(() => {
```

Comment:
This extra `useCallback` IIFE is redundant (and an antipattern). use the `useInfiniteQuery` hook directly. The hook already has memoization built in.